crates/mipsy_utils/src/expand.rs

```diff
@@ -7,10 +7,17 @@ pub fn expand_tilde<P: AsRef<Path> + ?Sized>(path: &'_ P) -> Cow<'_, Path> {
         return path.into();
     }
 
-    // TODO: support ~other_user/rest/of/path
-    if let Some(home) = home_dir() {
+    let home = if let Some(home) = home_dir() {
+        home
+    } else {
+        return path.into()
+    };
+
+    if path.starts_with("~/") {
         home.join(path.strip_prefix("~").unwrap()).into()
     } else {
-        path.into()
+        // TODO: support ~other_user/rest/of/path
+        // for now, assume this is a lone ~
```

Comment:
https://docs.rs/users might help

Reply:
This doesn't seem to provide the ability to retrieve a user's home directory, only their uid, name, and groups.
The current Rust solution seems to be to interface with the C function `getpwnam_r`, not sure why the Users crate doesn't have this

Reply:
wait nvm reading the source code it does provide this under OS_specific traits but this wasn't in the documentation

Reply:
There is also
https://crates.io/crates/shellexpand
But it looks to be unmaintained.